src/OpenLoco/Paint/PaintEntity.cpp

```diff
@@ -32,7 +33,153 @@ namespace OpenLoco::Paint
+                }
+                break;
+            }
+
+            case MiscThingType::redGreenCurrency: // 1
+            {
+                if (dpi->zoom_level > 1) // 004403C5-004403D0
+                {
+                    return;
+                }
+                MoneyEffect* moneyEffect = base->asRedGreenCurrency();
+                const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_green : StringIds::format_currency_expense_red_negative; // 004403D2-004403DF
+                uint32_t currencyAmount = abs(moneyEffect->amount);
+                uint32_t yOffsets = 0x4FAAC8 + moneyEffect->wiggle; // 004403E3
+
+                session.PaintFloatingMoneyEffect(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX); // 004403F7
+                break;
+            }
+
+            case MiscThingType::windowCurrency: // 2
+            {
+                if (dpi->zoom_level > 1) // 00440400-0044040B
+                {
+                    return;
+                }
+                MoneyEffect* moneyEffect = base->asWindowCurrency();
+                const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_in_company_colour : StringIds::format_currency_expense_in_company_colour_negative; // 0044040D-0044041A
+                uint32_t currencyAmount = abs(moneyEffect->amount);
+                uint32_t yOffsets = 0x4FAAC8 + moneyEffect->wiggle;                             // 0044041E-00440422
+                uint16_t companyColour = CompanyManager::getCompanyColour(moneyEffect->var_2E); // 00440428-0044042C
+                addr<0xE3F0A8, int16_t>() = companyColour;                                      // 00440434
+
+                session.PaintFloatingMoneyEffect(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX); // 00440445
+                break;
+            }
+
+            case MiscThingType::vehicleCrashParticle: // 3
+            {
+                if (dpi->zoom_level != 0) // 0044044E-00440459
+                {
+                    return;
+                }
+
+                VehicleCrashParticle* particle = base->asVehicleCrashParticle();
+
+                constexpr uint32_t IMAGE_TYPE_REMAP = (1 << 29);
+                constexpr uint32_t IMAGE_TYPE_REMAP_2_PLUS = (1u << 31);
+                static_assert((IMAGE_TYPE_REMAP | IMAGE_TYPE_REMAP_2_PLUS) == 0x0A0000000);
+
+                loco_global<int32_t[5], 0x4FAAB4> vehicle_particle_base_sprites;
+                uint32_t imageId = vehicle_particle_base_sprites[particle->crashedSpriteBase] + particle->frame / 256;                      // 0044045F-0044046A
+                imageId = imageId | (particle->colour[0] << 19) | (particle->colour[1] << 24) | IMAGE_TYPE_REMAP | IMAGE_TYPE_REMAP_2_PLUS; // 00440471-00440487
```

Comment:
`imageId = recolour2(imageId, particle->colour[0], particle->colour[1]);`

Reply:
I don't see that function in the code at the moment.

Reply:
oh doh it must be in my bogie branch
```
    uint32_t recolour2(uint32_t image, uint8_t colour1, uint8_t colour2)
    {
        return ImageIdFlags::remap | ImageIdFlags::remap2 | (colour1 << 19) | (colour2 << 24) | image;
    }```

Reply:
All right, implemented now as well (with an overloaded version).